ledger-core/v2/virtual/execute/execute.go

```diff
@@ -277,13 +277,15 @@ func (s *SMExecute) stepSaveNewObject(ctx smachine.ExecutionContext) smachine.St
 		_, prototype, memory = executionNewState.Activate()
 		action = func(state *object.SharedState) {
 			state.Info.SetDescriptor(&prototype, memory)
+			state.SetState(object.HasState)
 		}
 
 	case insolar.RequestSideEffectAmend:
 		_, prototype, memory = executionNewState.Amend()
 		action = func(state *object.SharedState) {
 			ctx.Log().Trace("applying new state")
 			state.Info.SetDescriptor(&prototype, memory)
+			state.SetState(object.HasState)
```

Comment:
Looks a little error-prone. Maybe this change should be made automatically in Set* methods?

Reply:
Yes. But
For example in VStateReport we will copy a lot of fields from incoming message to 'state'.
And object is ready only all of them are copied. In this case it's not clear hot to that automatically